### PR TITLE
[Improvement] Optimize the use of QuotaManager

### PR DESCRIPTION
**Pull request by smallzhongfeng**
### What changes were proposed in this pull request?

1. Optimize the use of `QuotaManager`, use `QuotaManager` when checking the number of apps, and decouple with `ApplicationManager`.
2. For #356  The start of `QuotaManager` is controlled by the configuration of the checker.

### Why are the changes needed?

More reasonable.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Added new ut.

### CODE REVIEW

**Comment by jerqi** [comment]
cc @xianjingfeng Could you help me review this patch?
If you feel that this patch is ok, you approve this pr and say `LGTM`. (`LGTM` means that look good to me)
企业微信截图_86ea0a12-15ab-45a3-b42b-6a9264dbada6

File: coordinator/src/main/java/org/apache/uniffle/coordinator/ApplicationManager.java [hunk 1]
```diff
@@ -75,10 +77,13 @@ public ApplicationManager(CoordinatorConf conf) {
       throw new UnsupportedOperationException("Unsupported selected storage strategy.");
     }
     expired = conf.getLong(CoordinatorConf.COORDINATOR_APP_EXPIRED);
-    QuotaManager quotaManager = new QuotaManager(conf);
-    this.currentUserAndApp = quotaManager.getCurrentUserAndApp();
-    this.appIdToUser = quotaManager.getAppIdToUser();
-    this.defaultUserApps = quotaManager.getDefaultUserApps();
+    if (conf.get(CoordinatorConf.COORDINATOR_ACCESS_CHECKERS).stream().map(String::trim).collect(Collectors.toList())
```
Comment (xianjingfeng) [inline coordinator/src/main/java/org/apache/uniffle/coordinator/ApplicationManager.java:80]:
```java
String quotaCheckerClass = AccessQuotaChecker.class.getCanonicalName();
for (String checker : conf.get(COORDINATOR_ACCESS_CHECKERS)) {
  if (quotaCheckerClass.equals(checker.trim())) {
    this.quotaManager = new QuotaManager(conf);
    this.currentUserAndApp = quotaManager.getCurrentUserAndApp();
    this.appIdToUser = quotaManager.getAppIdToUser();
    this.defaultUserApps = quotaManager.getDefaultUserApps();
    break;
  }
}
```
Do you think it's good to write like this?

File: coordinator/src/main/java/org/apache/uniffle/coordinator/ApplicationManager.java [hunk 2]
```diff
@@ -75,10 +77,13 @@ public ApplicationManager(CoordinatorConf conf) {
...
+        .contains(AccessQuotaChecker.class.getCanonicalName())) {
+      this.quotaManager = new QuotaManager(conf);
+      this.currentUserAndApp = quotaManager.getCurrentUserAndApp();
+      this.appIdToUser = quotaManager.getAppIdToUser();
+      this.defaultUserApps = quotaManager.getDefaultUserApps();
```
Comment (xianjingfeng) [inline coordinator/src/main/java/org/apache/uniffle/coordinator/ApplicationManager.java:85]: Can we remove `defaultUserApps` in this class?

File: coordinator/src/main/java/org/apache/uniffle/coordinator/QuotaManager.java [hunk 3]
```diff
@@ -53,13 +53,14 @@ public class QuotaManager {
   private final Map<String, Integer> defaultUserApps = Maps.newConcurrentMap();
 
   public QuotaManager(CoordinatorConf conf) {
-    this.quotaFilePath = conf.get(CoordinatorConf.COORDINATOR_QUOTA_DEFAULT_PATH);;
+    this.quotaFilePath = conf.get(CoordinatorConf.COORDINATOR_QUOTA_DEFAULT_PATH);
```
Comment (xianjingfeng) [inline coordinator/src/main/java/org/apache/uniffle/coordinator/QuotaManager.java:56]: If `quotaFilePath` is empty, should we throw an exception here  right now?

File: coordinator/src/main/java/org/apache/uniffle/coordinator/QuotaManager.java [hunk 4]
```diff
@@ -53,13 +53,14 @@ public class QuotaManager {
...
     final Long updateTime = conf.get(CoordinatorConf.COORDINATOR_QUOTA_UPDATE_INTERVAL);
     try {
       hadoopFileSystem = HadoopFilesystemProvider.getFilesystem(new Path(quotaFilePath), new Configuration());
     } catch (Exception e) {
       LOG.error("Cannot init remoteFS on path : " + quotaFilePath, e);
     }
+    LOG.warn("We will update the quota of users regularly.");
```
Comment (xianjingfeng) [inline coordinator/src/main/java/org/apache/uniffle/coordinator/QuotaManager.java:63]: Why we need this?

Comment (smallzhongfeng) [reply coordinator/src/main/java/org/apache/uniffle/coordinator/QuotaManager.java:63]: This is to prompt the user that QuotaManager starts normally.

File: coordinator/src/main/java/org/apache/uniffle/coordinator/QuotaManager.java (hunk 3 above)
Comment (smallzhongfeng) [reply coordinator/src/main/java/org/apache/uniffle/coordinator/QuotaManager.java:56]: If the quotaFilePath is empty, the quota for each user is the default value of rss.coordinator.quota.default.app.num. No exception is thrown, but logs can be added

File: coordinator/src/main/java/org/apache/uniffle/coordinator/ApplicationManager.java (hunk 2 above)
Comment (smallzhongfeng) [reply coordinator/src/main/java/org/apache/uniffle/coordinator/ApplicationManager.java:85]: Yes, this's not used in this class.

File: coordinator/src/main/java/org/apache/uniffle/coordinator/ApplicationManager.java (hunk 1 above)
Comment (smallzhongfeng) [reply coordinator/src/main/java/org/apache/uniffle/coordinator/ApplicationManager.java:80]: It's ok for me.

File: coordinator/src/main/java/org/apache/uniffle/coordinator/QuotaManager.java (hunk 3 above)
Comment (xianjingfeng) [reply coordinator/src/main/java/org/apache/uniffle/coordinator/QuotaManager.java:57]: I mean if quotaFilePath is empty, we should not try `getFilesystem` or create `ScheduledExecutorService`

File: coordinator/src/main/java/org/apache/uniffle/coordinator/QuotaManager.java (hunk 4 above)
Comment (xianjingfeng) [reply coordinator/src/main/java/org/apache/uniffle/coordinator/QuotaManager.java]: How about use LOG.info?

Comment (smallzhongfeng) [reply coordinator/src/main/java/org/apache/uniffle/coordinator/QuotaManager.java]: Updated.

**Review by xianjingfeng — APPROVED** [review]
LGTM

**Review by jerqi — APPROVED** [review]
LGTM, wait for CI. thanks @smallzhongfeng @xianjingfeng

**Comment by smallzhongfeng** [comment]
Thanks all. @jerqi @xianjingfeng

**Pull request merged**